File: app/src/main/java/com/master/kit/adapter/MainRecyclerAdapter.java
package com.master.kit.adapter;

import android.app.Activity;
import android.app.Dialog;
import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.master.kit.R;
import com.master.kit.bean.ScreenItemBean;

import tf.oof.com.service.base.BaseActivity;
import tf.oof.com.service.base.BaseSlidingActivity;
import tf.oof.com.service.base.BaseRecyclerAdapter;

import java.lang.reflect.InvocationTargetException;


/**
 * Created by master on 2017/9/25 0025.
 */

public class MainRecyclerAdapter extends BaseRecyclerAdapter<ScreenItemBean,MainRecyclerAdapter.RecyclerHolder> {
    LayoutInflater inflater;
    BaseActivity baseActivity;
    public MainRecyclerAdapter(BaseActivity context){
        this.baseActivity = context;
        inflater = LayoutInflater.from(context);
    }
    //LayoutInflater inflater = LayoutInflater.from(MainActivity.this);

    @Override
    public RecyclerHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = inflater.inflate(R.layout.item_recyclerview_main, parent, false);
        RecyclerHolder holder = new RecyclerHolder(view);
        return holder;
    }


    @Override
    public void onBindViewHolder(RecyclerHolder holder, final int position) {
        final ScreenItemBean bean = getItem(position);
        holder.describe.setText(bean.getDescribe());
        holder.name.setText(bean.getName()+"("+bean.getClz().getSimpleName()+")");
        holder.icon.setImageResource(bean.getIcon());
        if(bean.isIssue()){
            holder.issue.setVisibility(View.VISIBLE);
        }else{
            holder.issue.setVisibility(View.INVISIBLE);
        }

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (Activity.class.isAssignableFrom(bean.getClz())) {
                    baseActivity.startActivity(bean.getClz());
                } else if (Dialog.class.isAssignableFrom(bean.getClz())) {
                    try {
                        Dialog dialog = (Dialog) bean.getClz().getConstructor(Context.class).newInstance(baseActivity);
                        dialog.show();
                    } catch (InstantiationException e) {
                        e.printStackTrace();
                    } catch (IllegalAccessException e) {
                        e.printStackTrace();
                    } catch (InvocationTargetException e) {
                        e.printStackTrace();
                    } catch (NoSuchMethodException e) {
                        e.printStackTrace();
                    }
                }

            }
        });
    }
    public static class RecyclerHolder extends RecyclerView.ViewHolder {
        public TextView name;
        public TextView describe;
        public ImageView icon;
        public ImageView issue;
        public RecyclerHolder(View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.name);
            describe = itemView.findViewById(R.id.describe);
            icon = itemView.findViewById(R.id.icon);
            issue = itemView.findViewById(R.id.issue);
        }
    }
}

File: service/src/main/java/com/ooftf/service/base/adapter/MainRecyclerAdapter.kt
package com.ooftf.service.base.adapter

import android.graphics.Paint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.alibaba.android.arouter.launcher.ARouter
import com.amulyakhare.textdrawable.TextDrawable
import com.ooftf.service.R
import com.ooftf.service.base.BaseActivity
import com.ooftf.service.bean.ScreenItemBean
import com.ooftf.service.other.ColorGenerator


/**
 * Created by master on 2017/9/25 0025.
 */

open class MainRecyclerAdapter(private var baseActivity: BaseActivity, internal var stickyView: TextView) : CategoryRecyclerAdapter<ScreenItemBean, MainRecyclerAdapter.RecyclerHolder>() {
    override fun getCategory(position: Int) = getItem(position).category

    //LayoutInflater inflater = LayoutInflater.from(MainActivity.this);
    var inflater: LayoutInflater = LayoutInflater.from(baseActivity)

    override fun onCreateViewHolderSecondary(parent: ViewGroup, viewType: Int): RecyclerHolder {
        val view = inflater.inflate(R.layout.item_recyclerview_main, parent, false)
        return RecyclerHolder(view)
    }
    

    override fun onBindViewHolderSecondary(holder: RecyclerHolder, position: Int) {
        val bean = getItem(position)
        when {
            position == 0 -> holder.title.visibility = View.VISIBLE//第一个title必显示
            bean.category == getItem(position - 1).category -> holder.title.visibility = View.GONE//如果和上一个属于同一个类别，不显示
            else -> holder.title.visibility = View.VISIBLE//如果和上一个不属于同一个类别，显示
        }
        holder.title.text = bean.category
        holder.describe.text = bean.describe
        holder.name.text = bean.name + "(" + bean.clz + ")"
        if (bean.icon == R.drawable.logo_legacy || bean.icon == 0) {

            val drawable = TextDrawable.builder()
                    .beginConfig()
                    .fontSize(26)
                    .height(64)
                    .width(64)
                    .withBorder(4)
                    .endConfig()
                    .buildRound(bean.name.subSequence(0, 1).toString(), ColorGenerator.MATERIAL.randomColor)
            var pain = Paint()
            pain.isAntiAlias = true
            holder.icon.setLayerPaint(pain)
            holder.icon.setImageDrawable(drawable)
        } else {
            holder.icon.setImageResource(bean.icon)
        }


        if (bean.isIssue) {
            holder.issue.visibility = View.VISIBLE
        } else {
            holder.issue.visibility = View.INVISIBLE
        }

        holder.itemView.setOnClickListener {
            /*if (Activity::class.java.isAssignableFrom(bean.path)) {
                baseActivity.startActivity(bean.path)
            }*/
            ARouter.getInstance().build(bean.clz).navigation()
        }
    }

    class RecyclerHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var name: TextView = itemView.findViewById(R.id.name)
        var describe: TextView = itemView.findViewById(R.id.describe)
        var icon: ImageView = itemView.findViewById(R.id.icon)
        var issue: ImageView = itemView.findViewById(R.id.issue)
        var title: TextView = itemView.findViewById(R.id.stickyView)

    }

}
